examples/Planning/ForecastReach.php

```diff
@@ -0,0 +1,350 @@
+
+/**
+ * This example demonstrates how to interact with the ReachPlanService to find plannable
+ * locations and product codes, build a media plan, and generate a video ads reach forecast.
+ */
+class ForecastReach
+{
+    private const CUSTOMER_ID = 'INSERT_CUSTOMER_ID_HERE';
+    private const CURRENCY_CODE = 'USD';
+    // You can get a valid location ID from
+    // https://developers.google.com/adwords/api/docs/appendix/geotargeting or by calling
+    // ListPlannableLocations on the ReachPlanService.
+    private const LOCATION_ID = '2840'; // US
+    private const BUDGET_MICROS = 500000000000; // 500,000
+
+    public static function main()
+    {
+        // Either pass the required parameters for this example on the command line, or insert
+        // them into the constants above.
+        $options = (new ArgumentParser())->parseCommandArguments([
+            ArgumentNames::CUSTOMER_ID => GetOpt::REQUIRED_ARGUMENT
+        ]);
+
+        // Generate a refreshable OAuth2 credential for authentication.
+        $oAuth2Credential = (new OAuth2TokenBuilder())->fromFile()->build();
+
+        // Construct a Google Ads client configured from a properties file and the
+        // OAuth2 credentials above.
+        $googleAdsClient = (new GoogleAdsClientBuilder())
+            ->fromFile()
+            ->withOAuth2Credential($oAuth2Credential)
+            ->build();
+
+        try {
+            self::runExample(
+                $googleAdsClient,
+                $options[ArgumentNames::CUSTOMER_ID] ?: self::CUSTOMER_ID
+            );
+        } catch (GoogleAdsException $googleAdsException) {
+            printf(
+                "Request with ID '%s' has failed.%sGoogle Ads failure details:%s",
+                $googleAdsException->getRequestId(),
+                PHP_EOL,
+                PHP_EOL
+            );
+            foreach ($googleAdsException->getGoogleAdsFailure()->getErrors() as $error) {
+                /** @var GoogleAdsError $error */
+                printf(
+                    "\t%s: %s%s",
+                    $error->getErrorCode()->getErrorCode(),
+                    $error->getMessage(),
+                    PHP_EOL
+                );
+            }
+        } catch (ApiException $apiException) {
+            printf(
+                "ApiException was thrown with message '%s'.%s",
+                $apiException->getMessage(),
+                PHP_EOL
+            );
+        }
+    }
+
+    /**
+     * Runs the example.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     */
+    public static function runExample(GoogleAdsClient $googleAdsClient, int $customerId)
+    {
+        self::showPlannableLocations($googleAdsClient);
+
+        self::showPlannableProducts($googleAdsClient);
+
+        self::forecastManualMix($googleAdsClient, $customerId);
+
+        self::forecastSuggestedMix($googleAdsClient, $customerId);
```

Comment:
I don't think we need space between these lines, unless you want to explain something (but descriptions can be found in each method already).